### PR TITLE
#159965265 allow users to reset their password 

### PR DESCRIPTION
**Pull request by meshack-mbuvi**
#### What does this PR do?
Have reset password feature implemented on the frontend

#### Description of Task to be completed
Have components which users enter details to reset their password

#### How should this be manually tested
$ git clone https://github.com/andela/mag-ah-frontend.git
$ git checkout ft-reset-password-159965265
$ cd mag-ah-frontend
$ npm install
$ npm start

Navigate to ```http://localhost:3000/login```
$ click forgot password button on the form
$ Fill out your email address on the dialog and click *Send Reset link*
$ Follow resulting instruction displayed on the form

#### Any background context you want to provide?
Regardless of who the person is, forgetting password is usually common and therefore there is a need to provide a way by which users can recover their accounts in any given application.
#### What are the relevant pivotal tracker stories?
[User should be presented with a reset password button](https://www.pivotaltracker.com/story/show/159965265)
#### Screenshots (if appropriate)
Forgot password button on login modal
image

An email Input field and button to generate `password reset link`
image

A page to where users are redirected from their email
image

### CODE REVIEW

File: src/__tests__/resetpassword/ForgotPassword.test.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const email = "test@gmail.com";
+  it("should dispatch an action to generate email address", () => {
+    const expectedAction = [
+      {
+        type: FORGOT_PWD,
+        responseData: "test@gmail.com"
+      }
+    ];
+    //   configure Mock store
+    const store = mockStore({ response: { message: "check your email" } });
```
Comment (pabishai) [inline src/__tests__/resetpassword/ForgotPassword.test.jsx]: @meshack-mbuvi I suggest you declare the store globally and then clear actions in the `afterEach()` function

File: src/redux/actions/ForgotPasswordAction.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import axios from "axios";
+import * as ActionTypes from "../action_types";
+
+/* request to change password
```
Comment (pabishai) [inline src/redux/actions/ForgotPasswordAction.js]: @kimotho-njoki Should this be above the function? I thought comments should be inside the function? @meshack-mbuvi

**Review by pabishai — COMMENTED** [review]
@meshack-mbuvi The modal user interface does seem to match what the login one has. This will change once login is merged so it may not be an issue.

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 1 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx]: Your feedback has already been taken into account.

File: src/components/ResetPassword.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        password: "",
+        confirm_password: ""
+      },
+      info: true
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
```
Comment (micahoriaso) [inline src/components/ResetPassword.jsx]: you could refactor this to:
``` 
handleChange = e => {
....
}
````
so that you can get rid of the
```this.handleChange = this.handleChange.bind(this)``` 
in the constructor

Comment (micahoriaso) [reply src/components/ResetPassword.jsx]: Do the same for ```handleSubmit```

**Review by micahoriaso — CHANGES_REQUESTED** [review]

File: src/components/ResetPassword.jsx (hunk 3 above)
Comment (meshack-mbuvi) [reply src/components/ResetPassword.jsx]: @micahoriaso This has been taken into account as at now

File: src/redux/actions/ForgotPasswordAction.js (hunk 2 above)
Comment (meshack-mbuvi) [reply src/redux/actions/ForgotPasswordAction.js]: Look at this file https://react-styleguidist.js.org/docs/documenting.html at @PAbishai

**Review by micahoriaso — APPROVED** [review]

File: src/redux/actions/ForgotPasswordAction.js (hunk 2 above)
Comment (kimotho-njoki) [reply src/redux/actions/ForgotPasswordAction.js]: Above!

**Comment by kimotho-njoki** [comment]
@meshack-mbuvi First things first, the modal is waaaaaay too up on the page, it should be dancing somewhere in the middle... Space out the input fields as well to make the modal a little longer.
The title is not centered as well.

File: src/__tests__/resetpassword/ForgotPassword.test.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+const mockStore = configureMockStore(middlewares);
+//   configure Mock store
+const store = mockStore({ response: { message: "check your email" } });
+describe("user action", () => {
+  beforeEach(() => {
+    moxios.install();
+  });
+  afterEach(() => {
+    moxios.uninstall();
+  });
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ForgotPassword.test.jsx:26]: empty line after line 22

File: src/__tests__/resetpassword/ForgotPassword.test.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  const email = "test@gmail.com";
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ForgotPassword.test.jsx]: Why is this here and not above `describe()`? Talking about const email

File: src/__tests__/resetpassword/ForgotPassword.test.jsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      {
+        type: FORGOT_PWD,
+        responseData: "test@gmail.com"
+      }
+    ];
+
+    store.dispatch(forgotPassword(email));
+    expect(store.getActions()).toEqual(expectedAction);
+  });
+});
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ForgotPassword.test.jsx:39]: Empty line after line 35, keep your work clean

File: src/__tests__/resetpassword/ForgotPassword.test.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  const dispatch = jest.fn();
+  const props = { dispatch, message: {} };
+  const wrapper = shallow(<ForgotPassword {...props} />);
+
+  it("renders without crashing", () => {
+    expect(wrapper.find("form.forgotPassword").exists()).toBe(true);
+  });
+  it("renders an email input", () => {
+    expect(wrapper.find("#email").length).toEqual(1);
+  });
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ForgotPassword.test.jsx:50]: An empty line after every `it()`

File: src/__tests__/resetpassword/ForgotPassword.test.jsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        responseData: "test@gmail.com"
+      }
+    ];
+
+    store.dispatch(forgotPassword(email));
+    expect(store.getActions()).toEqual(expectedAction);
+  });
+});
+describe("forgotPassword component", () => {
+  const dispatch = jest.fn();
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ForgotPassword.test.jsx]: Is dispatch a prop?

File: public/index.html [hunk 9]
```diff
@@ -36,5 +43,6 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-  </body>
-</html>
+</body>
+
+</html>
```
Comment (kimotho-njoki) [inline public/index.html]: No trailing empty line after line 48?

File: src/__tests__/resetpassword/ForgotPassword.test.jsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  const props = { dispatch, message: {} };
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ForgotPassword.test.jsx]: Props are defined at the beginning of this page. Right before the first `describe()`.

const props = {
   dispatch: jest.fn(),
   ...
}

Comment (kimotho-njoki) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx]: And how are you declaring it here but defining it right above?

File: src/__tests__/resetpassword/ResetPassword.test.jsx [hunk 11]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import Enzyme, { shallow } from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import { ResetPassword } from "../../components/auth/ResetPassword";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("forgotPassword component", () => {
+  const dispatch = jest.fn();
+  const props = { dispatch, message: {} };
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ResetPassword.test.jsx]: Same case for props here

File: src/__tests__/resetpassword/ResetPassword.test.jsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        .find("form.resetPassword")
+        .exists()
+    ).toBe(true);
+  });
+  it("renders an submit button", () => {
+    expect(shallow(<ResetPassword {...props} />).find("#submit").length).toBe(
+      1
+    );
+  });
+  describe("Component input fields", () => {
```
Comment (kimotho-njoki) [inline src/__tests__/resetpassword/ResetPassword.test.jsx:24]: You really hate empty lines. Empty line after each `it()`. Esp before a `describe()`

File: src/components/auth/AuthButton.jsx [hunk 13]
```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import Button from "../../views/Button/index";
+import { Logout } from "../../redux/actions/loginAction";
+
+class AuthButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onLogout = this.onLogout.bind(this);
```
Comment (kimotho-njoki) [inline src/components/auth/AuthButton.jsx]: I think you received feedback on this from @micahoriaso . Let's get rid of this bind statements and the unnecessary constructor. Read on https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56

File: src/components/auth/AuthButton.jsx [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    event.preventDefault();
+    dispatch(Logout());
+  }
+
+  render() {
+    const { isAuthenticated } = this.props;
+
+    return (
+      <div>
+        {isAuthenticated ? (
```
Comment (kimotho-njoki) [inline src/components/auth/AuthButton.jsx:19]: This component looks exact to a component I saw on @mikenthiwa's PR??? Anyway, check out the arrangement for a long ternary operator right here https://github.com/airbnb/javascript/issues/1281, the first one with comment //better

File: src/components/auth/ForgotPasswordComp.jsx [hunk 15]
```diff
@@ -0,0 +1,111 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import Button from "../../views/Button";
+import { TextInput } from "../../views/Form";
+import { handleForgotPassword } from "../../redux/actions/ForgotPasswordAction";
+
+export class ForgotPassword extends Component {
+  constructor(props) {
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx]: This constructor is unnecessary as the bind statements have been removed

File: src/components/auth/ForgotPasswordComp.jsx [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+            data-target="#ahSignInModal"
+          >
+            Back to login form
+          </a>
+        </span>
+      </div>
+    );
+  }
+}
+ForgotPassword.propTypes = {
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx:102]: Empty line after 101, 106, 109 and 110

File: src/components/auth/ForgotPasswordComp.jsx [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+  dispatch: PropTypes.func.isRequired,
+  message: PropTypes.objectOf(PropTypes.string).isRequired,
+  error: PropTypes.objectOf(PropTypes.object)
+};
+ForgotPassword.defaultProps = {
+  error: { data: { errors: {} }, message: { message: "" } }
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx]: Refactor to
error: {
data: { errors: {} },
...
}

**Comment by meshack-mbuvi** [comment]
@kimotho-njoki I will be implementing your feedback one by one. Just give me as much as you can

File: src/components/auth/ForgotPasswordComp.jsx [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+};
+const mapStateToProps = ({ resetPasswordReducer }) => resetPasswordReducer;
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx]: What do you need from this resetPasswordReducer? Refactor to

const mapStateToProps = ({ resetPasswordReducer }) => {
 const { everythingYouNeed/thePropsThatIs } = resetPasswordReducer;
return {
everyPropYouStatedYouNeedUpThere
};
};

File: src/components/auth/LoginForm.jsx [hunk 19]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { TextInput } from "../../views/Form/index";
+
+const LoginForm = ({ error, password, handleChange, email }) => (
+  <form>
+    {error && (
+      <p className="ah-err-msg">
+        {error.email ? "Email is required" : ""}
+        {error ? error.error : ""}
```
Comment (kimotho-njoki) [inline src/components/auth/LoginForm.jsx:10]: Inline ternaries are okay here, since this is a short ternary

File: src/components/auth/LoginForm.jsx [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+LoginForm.propTypes = {
+  email: PropTypes.string.isRequired,
+  password: PropTypes.string.isRequired,
+  error: PropTypes.shape(),
+  handleChange: PropTypes.func.isRequired
+};
+
+LoginForm.defaultProps = {
+  error: {}
+};
```
Comment (kimotho-njoki) [inline src/components/auth/LoginForm.jsx:55]: Empty line after 53

File: src/components/auth/ResetPassword.jsx [hunk 21]
```diff
@@ -0,0 +1,135 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { Redirect } from "react-router-dom";
+import { handleResetPassword } from "../../redux/actions/ResetPasswordAction";
+import SubmitButton from "../../views/Form/SubmitButton";
+import { TextInput } from "../../views/Form";
+
+export class ResetPassword extends Component {
+  constructor(props) {
```
Comment (kimotho-njoki) [inline src/components/auth/ResetPassword.jsx]: Again, useless constructor

File: src/components/auth/ResetPassword.jsx [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+    const { user } = this.state;
+    const { dispatch } = this.props;
+    dispatch(handleResetPassword(user));
+  };
+
+  render() {
+    const { user, info } = this.state;
+    const { password, confirmPassword } = user;
+    const { error, message } = this.props;
+    //   Redirect user when we have a successful message received from server
```
Comment (kimotho-njoki) [inline src/components/auth/ResetPassword.jsx]: Remove unnecessary comments, keep your code clean

File: src/components/auth/ResetPassword.jsx [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
+          <div className="m-auto col-3">
+            <form onSubmit={this.handleSubmit} className="resetPassword">
+              {message && (
+                <p className="text-center">{message ? message.message : ""}</p>
+              )}
+
+              <div className="form-group">
+                <div className="form-group mb-4">
+                  <h4>
+                    {/* Display helper message when no error is set */}
```
Comment (kimotho-njoki) [inline src/components/auth/ResetPassword.jsx]: Again, Unnecessary comment

File: src/components/auth/ResetPassword.jsx [hunk 24]
```diff
@@ -0,0 +1,135 @@
...
+                    {info && (
+                      <p className="text-center">
+                        {error ? (
```
Comment (kimotho-njoki) [inline src/components/auth/ResetPassword.jsx]: Wrong arrangement of ternary, the one on line 76 is the correct one

File: src/components/auth/ResetPassword.jsx [hunk 25]
```diff
@@ -0,0 +1,135 @@
...
+                          <span className="text-danger">
+                            {error.data.errors.error
+                              ? error.data.errors.error
```
Comment (kimotho-njoki) [inline src/components/auth/ResetPassword.jsx]: Why not use &&, you not doing anything on the `else` part? Nested ternaries are frowned upon, they are not readable.

File: src/components/auth/ResetPassword.jsx [hunk 26]
```diff
@@ -0,0 +1,135 @@
...
+}
+ResetPassword.propTypes = {
+  dispatch: PropTypes.func.isRequired,
+  message: PropTypes.objectOf(PropTypes.string).isRequired,
+  error: PropTypes.objectOf(PropTypes.object)
+};
+
+ResetPassword.defaultProps = {
+  error: { data: { errors: {} }, message: { message: "" } }
+};
```
Comment (kimotho-njoki) [inline src/components/auth/ResetPassword.jsx:129]: Empty line after 133

File: src/components/auth/ResetPassword.jsx [hunk 27]
```diff
@@ -0,0 +1,135 @@
...
+const mapStateToProps = ({ resetPasswordReducer }) => resetPasswordReducer;
```
Comment (kimotho-njoki) [inline src/components/auth/ResetPassword.jsx]: Refactor same way I said up there 👆

File: src/components/auth/login.jsx [hunk 28]
```diff
@@ -0,0 +1,89 @@
...
+import loginAction from "../../redux/actions/loginAction";
+import Button from "../../views/Button/index";
+import LoginForm from "./LoginForm";
+
+class LoginPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: "", password: "" };
+    this.handleChange = this.handleChange.bind(this);
+    this.onLogin = this.onLogin.bind(this);
```
Comment (kimotho-njoki) [inline src/components/auth/login.jsx]: I can still see bind statements. Get rid of them as well as the constructor.

File: src/components/auth/login.jsx [hunk 29]
```diff
@@ -0,0 +1,89 @@
...
+    return (
+      <div className="modal-body text-center">
+        <LoginForm
+          handleChange={this.handleChange}
+          email={email}
+          password={password}
+          error={error}
+        />
+        <div className="col-9 m-auto">
+          {/* Show modal for account recovery */}
```
Comment (kimotho-njoki) [inline src/components/auth/login.jsx]: Unnecessary comment

File: src/components/auth/login.jsx [hunk 30]
```diff
@@ -0,0 +1,89 @@
...
+        </div>
+      </div>
+    );
+  }
+}
+
+LoginPage.propTypes = {
+  dispatch: PropTypes.func.isRequired,
+  error: PropTypes.shape()
+};
```
Comment (kimotho-njoki) [inline src/components/auth/login.jsx:81]: Extra empty line after 85, 88

File: src/index.css [hunk 31]
```diff
@@ -1,14 +1,26 @@
...
+}
+
+.custom-file-label {
+    background-color: transparent;
+}
+
+.form-control:focus {
+    border-bottom: solid 1px;
+    box-shadow: None;
+}
```
Comment (kimotho-njoki) [inline src/index.css]: Wheres the trailing empty line on this page?

File: src/redux/actions/ForgotPasswordAction.js [hunk 32]
```diff
@@ -0,0 +1,41 @@
...
+  type: ActionTypes.FORGOT_PWD,
+  responseData
+});
+
+/* error from forgot password request
+*
+* @param {string} error
+* @return {object} type and responseData,{boolean} info
+*/
+
```
Comment (kimotho-njoki) [inline src/redux/actions/ForgotPasswordAction.js]: Remove this extra line after the comment. And your asterisks are placed incorrectly, please ask me in person.

File: src/views/App/App.scss [hunk 33]
```diff
@@ -1,33 +1,225 @@
...
+.ah-google-btn {
+    background-color: $google-color;
+    color: #ffffff !important;
+}
+
+.modal-body .ah-err-msg {
+    color: red;
+}
+
+#snackbar {
```
Comment (kimotho-njoki) [inline src/views/App/App.scss]: Why all these comments?

File: src/views/App/App.scss [hunk 34]
```diff
@@ -1,33 +1,225 @@
...
 }
+
+
+/* Show the snackbar when clicking on a button (class added with JavaScript) */
+
+#snackbar.show {
+    visibility: visible;
+    /* Show the snackbar */
+    /* Add animation: Take 0.5 seconds to fade in and out the snackbar.
+ However, delay the fade out process for 2.5 seconds */
```
Comment (kimotho-njoki) [inline src/views/App/App.scss]: Please get rid of the comments, keep your code clean

File: src/views/App/App.scss [hunk 35]
```diff
@@ -1,33 +1,225 @@
...
+    -webkit-animation: fadein 0.5s, fadeout 0.5s 2.5s;
+    animation: fadein 0.5s, fadeout 0.5s 2.5s;
+}
+
+
+/* Animations to fade the snackbar in and out */
```
Comment (kimotho-njoki) [inline src/views/App/App.scss]: Delete.

File: src/views/Form/TextInput.jsx [hunk 36]
```diff
@@ -0,0 +1,45 @@
...
+        name={name}
+        onChange={onChange}
+        onBlur={onBlur}
+        placeholder={placeholder}
+        required={required}
+      />
+      {error && <div className="ah-input-error pt-1 text-danger">{error}</div>}
+    </div>
+  </div>
+);
```
Comment (kimotho-njoki) [inline src/views/Form/TextInput.jsx]: Extra empty line after 28, 38

File: src/components/auth/ForgotPasswordComp.jsx (hunk 18 above)
Comment (meshack-mbuvi) [reply src/components/auth/ForgotPasswordComp.jsx]: will do that

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 5 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx]: I have moved it up.

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 6 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx:39]: sorted out

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 7 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx:50]: Done

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 8 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx]: Yes, it is. Without passing it, the tests fail

File: public/index.html (hunk 9 above)
Comment (meshack-mbuvi) [reply public/index.html]: Oh my... the empty line is removed whenever I save. Let me check how to configure my editor to avoid this.

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 10 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx]: I have moved them up. Let me know your take.

File: src/__tests__/resetpassword/ResetPassword.test.jsx (hunk 11 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ResetPassword.test.jsx]: Moved them up

File: src/__tests__/resetpassword/ResetPassword.test.jsx (hunk 12 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ResetPassword.test.jsx]: No, I don't. Its my editor

File: src/components/auth/AuthButton.jsx (hunk 13 above)
Comment (meshack-mbuvi) [reply src/components/auth/AuthButton.jsx]: Done it already

File: src/components/auth/AuthButton.jsx (hunk 14 above)
Comment (meshack-mbuvi) [reply src/components/auth/AuthButton.jsx:19]: They are the one handling it. The formatting for this section is being handled by @mikenthiwa and @Bileonaire I was only interested in having my `forgot password` on the login form.

File: src/components/auth/ForgotPasswordComp.jsx (hunk 15 above)
Comment (meshack-mbuvi) [reply src/components/auth/ForgotPasswordComp.jsx]: Yep. I have taken care of it now.

File: src/components/auth/LoginForm.jsx (hunk 20 above)
Comment (meshack-mbuvi) [reply src/components/auth/LoginForm.jsx:55]: Done. Let me know whether the blank line on line 45 above is necessary. I have added it to improve readability.

File: src/components/auth/LoginForm.jsx (hunk 19 above)
Comment (meshack-mbuvi) [reply src/components/auth/LoginForm.jsx:10]: Ok

File: src/components/auth/ResetPassword.jsx (hunk 22 above)
Comment (meshack-mbuvi) [reply src/components/auth/ResetPassword.jsx]: Done

File: src/components/auth/ResetPassword.jsx (hunk 23 above)
Comment (meshack-mbuvi) [reply src/components/auth/ResetPassword.jsx]: okay

File: src/components/auth/ResetPassword.jsx (hunk 24 above)
Comment (meshack-mbuvi) [reply src/components/auth/ResetPassword.jsx]: Fixed

File: src/components/auth/ResetPassword.jsx (hunk 25 above)
Comment (meshack-mbuvi) [reply src/components/auth/ResetPassword.jsx]: Eliminating them

File: src/components/auth/ResetPassword.jsx (hunk 26 above)
Comment (meshack-mbuvi) [reply src/components/auth/ResetPassword.jsx:129]: Done

File: src/components/auth/ResetPassword.jsx (hunk 27 above)
Comment (meshack-mbuvi) [reply src/components/auth/ResetPassword.jsx]: Done

File: src/components/auth/login.jsx (hunk 28 above)
Comment (meshack-mbuvi) [reply src/components/auth/login.jsx]: Done

File: src/components/auth/login.jsx (hunk 29 above)
Comment (meshack-mbuvi) [reply src/components/auth/login.jsx]: Removed

File: src/components/auth/login.jsx (hunk 30 above)
Comment (meshack-mbuvi) [reply src/components/auth/login.jsx:81]: Done

File: src/index.css (hunk 31 above)
Comment (meshack-mbuvi) [reply src/index.css]: Inserted

File: src/redux/actions/ForgotPasswordAction.js (hunk 32 above)
Comment (meshack-mbuvi) [reply src/redux/actions/ForgotPasswordAction.js]: I have corrected them after the session

File: src/views/App/App.scss (hunk 33 above)
Comment (meshack-mbuvi) [reply src/views/App/App.scss]: Removed

File: src/views/App/App.scss (hunk 34 above)
Comment (meshack-mbuvi) [reply src/views/App/App.scss]: Done

File: src/views/App/App.scss (hunk 35 above)
Comment (meshack-mbuvi) [reply src/views/App/App.scss]: Done

File: src/views/Form/TextInput.jsx (hunk 36 above)
Comment (meshack-mbuvi) [reply src/views/Form/TextInput.jsx]: Done

File: src/components/auth/AuthButton.jsx (hunk 13 above)
Comment (kimotho-njoki) [reply src/components/auth/AuthButton.jsx]: I can still see bind...I can still see the constructor... what's going on?

File: src/components/auth/ForgotPasswordComp.jsx [hunk 37]
```diff
@@ -0,0 +1,120 @@
...
+          <div className="text-center">
+            <SubmitButton
+              className="btn btn-primary mb-4"
+              label="Generate Reset Link"
+              id="reset"
+            />
+          </div>
+        </form>
+        <span>
+          {/* Show modal for account recovery */}
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx]: Why is this comment still here?

File: src/components/auth/ForgotPasswordComp.jsx [hunk 38]
```diff
@@ -0,0 +1,120 @@
...
+  render() {
+    const { message, error } = this.props;
+    const { user } = this.state;
+    const { email } = user;
+    return (
+      <div className="m-auto col-sm-12">
+        <form onSubmit={this.handleSubmit} className="forgotPassword">
+          {message && (
+            <p className="text-center text-info info">
+              {/* Display message to user on successful link generation */}
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx]: And this comment?

File: src/components/auth/login.jsx (hunk 28 above)
Comment (kimotho-njoki) [reply src/components/auth/login.jsx]: What is done? I can still see a bind statement?

File: src/redux/actions/ForgotPasswordAction.js [hunk 39]
```diff
@@ -0,0 +1,48 @@
...
+  error
+});
+
+/**
+ * send request to generate reset password link
+ *
+ * @param {object} email
+ * @return {function} dispatch
+ */
+
```
Comment (kimotho-njoki) [inline src/redux/actions/ForgotPasswordAction.js]: Remove this extra line after the comment. Line 34, line 22 and the rest. Attention to detail.

File: src/views/App/App.scss [hunk 40]
```diff
@@ -1,33 +1,189 @@
...
+.btn-link:hover {
+    color: $ah-btn-link-color;
+    text-decoration: none;
+}
+
+.btn-outline-success:hover {
+    border: 1px solid $ah-btn-link-color;
+}
+
+// Card
```
Comment (kimotho-njoki) [inline src/views/App/App.scss]: Comments. Delete.

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 4 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx:26]: Done

File: src/__tests__/resetpassword/ForgotPassword.test.jsx (hunk 6 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ForgotPassword.test.jsx]: sorted out

File: src/__tests__/resetpassword/ResetPassword.test.jsx (hunk 12 above)
Comment (meshack-mbuvi) [reply src/__tests__/resetpassword/ResetPassword.test.jsx:24]: I don't hate them; they hate me. I have befriended them

File: src/components/auth/ForgotPasswordComp.jsx (hunk 16 above)
Comment (meshack-mbuvi) [reply src/components/auth/ForgotPasswordComp.jsx:102]: inserted

File: src/components/auth/ForgotPasswordComp.jsx [hunk 41]
```diff
@@ -0,0 +1,128 @@
...
+    const { message, error } = this.props;
+    const { user } = this.state;
+    const { email } = user;
+
+    return (
+      <div className="m-auto col-sm-12">
+        <form onSubmit={this.handleSubmit} className="forgotPassword">
+          {message && (
+            <p className="text-center text-info info">
+              {message.message ? (
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx:40]: Refactor this to 

message.message
? blahblah
:  blahblah

for readability

File: src/redux/actions/ResetPasswordAction.js [hunk 42]
```diff
@@ -0,0 +1,47 @@
...
+  error
+});
+
+/**
+ * reset user password
+ *
+ * @param {string} responseData
+ * @return {object} type and responseData
+ */
+
```
Comment (kimotho-njoki) [inline src/redux/actions/ResetPasswordAction.js]: Attention to detail @meshack-mbuvi. When you receive feedback regarding one page, ensure you check all your other pages

File: src/redux/actions/ResetPasswordAction.js [hunk 43]
```diff
@@ -0,0 +1,47 @@
...
+import * as ActionTypes from "../action_types";
+import API_BASE_URL from "./index";
+
+/**
+ * error from forgot password request
+ *
+ * @param {string} error
+ * @return {object} type and error
+ */
+
```
Comment (kimotho-njoki) [inline src/redux/actions/ResetPasswordAction.js]: You forgot to delete the empty lines on this page

File: src/redux/actions/ResetPasswordAction.js (hunk 42 above)
Comment (meshack-mbuvi) [reply src/redux/actions/ResetPasswordAction.js]: Point noted. I am fighting with backend bugs here but am managing

File: src/components/auth/ForgotPasswordComp.jsx (hunk 41 above)
Comment (meshack-mbuvi) [reply src/components/auth/ForgotPasswordComp.jsx:40]: I get linting errors when I do as you have requested

File: src/redux/actions/ResetPasswordAction.js (hunk 43 above)
Comment (meshack-mbuvi) [reply src/redux/actions/ResetPasswordAction.js]: I have deleted it.
